Analysis/Tasks/pidTOFqa.cxx

```diff
@@ -0,0 +1,207 @@
+      binp[i] = TMath::Exp(TMath::Log(10) * (lmin + i * ldelta));
+    }
+    binp[nbins] = max + 1;
+    h->GetXaxis()->Set(nbins, binp);
+  }
+
+  template <uint8_t i>
+  void addParticleHistos()
+  {
+    // NSigma
+    histos.add(hnsigmaMC[i].data(), Form("True %s;#it{p}_{T} (GeV/#it{c});N_{#sigma}^{TOF}(%s)", pT[i], pT[pid_type]), HistType::kTH2F, {{nBinsP, MinP, MaxP}, {2000, -30, 30}});
+    makelogaxis(histos.get<TH2>(HIST(hnsigmaMC[i])));
+    histos.add(hnsigmaMCprm[i].data(), Form("True Primary %s;#it{p}_{T} (GeV/#it{c});N_{#sigma}^{TOF}(%s)", pT[i], pT[pid_type]), HistType::kTH2F, {{nBinsP, MinP, MaxP}, {2000, -30, 30}});
+    makelogaxis(histos.get<TH2>(HIST(hnsigmaMCprm[i])));
+    histos.add(hnsigmaMCsec[i].data(), Form("True Secondary %s;#it{p}_{T} (GeV/#it{c});N_{#sigma}^{TOF}(%s)", pT[i], pT[pid_type]), HistType::kTH2F, {{nBinsP, MinP, MaxP}, {2000, -30, 30}});
+    makelogaxis(histos.get<TH2>(HIST(hnsigmaMCsec[i])));
+  }
+
+  void init(o2::framework::InitContext&)
+  {
+    histos.add(hnsigma[pid_type].data(), Form(";#it{p}_{T} (GeV/#it{c});N_{#sigma}^{TOF}(%s)", pT[pid_type]), HistType::kTH2F, {{nBinsP, MinP, MaxP}, {2000, -30, 30}});
+    makelogaxis(histos.get<TH2>(HIST(hnsigma[pid_type])));
+    histos.add(hnsigmaprm[pid_type].data(), Form("Primary;#it{p}_{T} (GeV/#it{c});N_{#sigma}^{TOF}(%s)", pT[pid_type]), HistType::kTH2F, {{nBinsP, MinP, MaxP}, {2000, -30, 30}});
+    makelogaxis(histos.get<TH2>(HIST(hnsigmaprm[pid_type])));
+    histos.add(hnsigmasec[pid_type].data(), Form("Secondary;#it{p}_{T} (GeV/#it{c});N_{#sigma}^{TOF}(%s)", pT[pid_type]), HistType::kTH2F, {{nBinsP, MinP, MaxP}, {2000, -30, 30}});
+    makelogaxis(histos.get<TH2>(HIST(hnsigmasec[pid_type])));
+    addParticleHistos<0>();
+    addParticleHistos<1>();
+    addParticleHistos<2>();
+    addParticleHistos<3>();
+    addParticleHistos<4>();
+    addParticleHistos<5>();
+    addParticleHistos<6>();
+    addParticleHistos<7>();
+    addParticleHistos<8>();
+  }
+
+  template <uint8_t pidIndex, typename T, typename TT>
+  void fillNsigma(const T& track, const TT& mcParticles, const float& nsigma)
+  {
+    if (abs(track.label().pdgCode()) == PDGs[pidIndex]) {
+      histos.fill(HIST(hnsigmaMC[pidIndex]), track.pt(), nsigma);
+
+      if (MC::isPhysicalPrimary(mcParticles, track.label())) { // Selecting primaries
+        histos.fill(HIST(hnsigmaMCprm[pidIndex]), track.pt(), nsigma);
+      } else {
+        histos.fill(HIST(hnsigmaMCsec[pidIndex]), track.pt(), nsigma);
+      }
+    }
+  }
+
+  void process(aod::Collision const& collision,
+               soa::Join<aod::Tracks, aod::TracksExtra, aod::pidRespTOF, aod::McTrackLabels> const& tracks,
+               aod::McParticles& mcParticles)
+  {
+    const float collisionTime_ps = collision.collisionTime() * 1000.f;
+    for (auto t : tracks) {
+      //
+      if (t.tofSignal() < 0) { // Skipping tracks without TOF
+        continue;
+      }
+      float nsigma = -999.f;
+      if (pid_type == 0) {
```

Comment:
Doesn't this have to be constrexpr?

Reply:
Indeed, however it worked without. I fixed it in the new version

Reply:
Sure, but it is done run time and therefore much slower...